src/module/mutations/setAllMutation.js

```diff
@@ -0,0 +1,22 @@
+import Vue from 'vue'
+
+/**
+ * Proxy for setting Resource Objects on a collection module
+ *
+ * The `Vuex.commit`-Syntax is
+ *
+ * `commit('module/set', receivedCollectionObject)`
+ *
+ * @param {Vuex.Store} store
+ * @param {resource.Builder} resourceBuilder
+ */
+export function setAllMutation (resourceBuilder) {
+  return new Proxy(() => {}, {
+    apply (target, thisArg, [state, payload]) {
+      const settablePayload = resourceBuilder.build(payload)
+
+      Vue.set(state, 'items', settablePayload)
+      Vue.set(state, 'initial', settablePayload)
```

Comment:
We will loose cached items from other pages, that is the current behaviour and subject to change. 

Single items changed but not pushed are a concern, but also not any more than previously, as this just does in bulk, what `setMutation` used to do in a loop.

Yeah. `Vue.set` probably isn't needed. But that probably vs the non-existence of a test to me means it stays because it works and the performance impact of this whole mutation is negligible now.

Reply:
I would suggest something like this: `this.state.items = {...this.state.items, ...settablePayload}`. Thus it would only override items that were actually requested during the call and we would have no change of behaviour compared to the current solution.

Reply:
ah. no, let me correct that. `setMutation` did not do that in a loop but the calling `setResourceObjectsForModule` did. So still no change to the previous behaviour in my opinion.

Reply:
What I am wondering is whether the initial variant of the object really needs to be functionalized. But I would rather refactor that some time later. I think those can just be frozen and recreated as functionalized on reset.